Source/WebCore/svg/SVGImageElement.cpp

```diff
@@ -64,20 +64,7 @@ Ref<SVGImageElement> SVGImageElement::create(const QualifiedName& tagName, Docum
 
 CachedImage* SVGImageElement::cachedImage() const
 {
-    const RenderImageResource* resource = nullptr;
-#if ENABLE(LAYER_BASED_SVG_ENGINE)
-    if (auto* renderer = dynamicDowncast<RenderSVGImage>(this->renderer()); renderer && renderer->imageResource().cachedImage())
-        resource = &renderer->imageResource();
-#endif
-    if (!resource) {
-        if (auto* renderer = dynamicDowncast<LegacyRenderSVGImage>(this->renderer()); renderer && renderer->imageResource().cachedImage())
-            resource = &renderer->imageResource();
-    }
-
-    if (!resource)
-        return nullptr;
-
-    return resource->cachedImage();
+    return m_imageLoader.image();
```

Comment:
I am not sure about this change. We have been returning the cached image only if the renderer is attached to the element even if it is loaded. Is there a reason for changing this now.

Anyway I think this part is not related to this patch. I think it can be done as a separate clean-up for SVG code.

Reply:
The reason is that all of the relevant WPTs for this are using an SVGImageElement that isn't inserted into the Document.

If we don't make this change, then we don't pass any new tests, and arguably we wouldn't add much value for real sites either.